cli/internal/context/context.go

```diff
@@ -146,34 +146,26 @@ func WithGraph(rootpath string, config *config.Config) Option {
 			return fmt.Errorf("could not resolve workspaces: %w", err)
 		}
 
-		spaces, err := c.PackageManager.GetWorkspaceGlobs(rootpath)
-
-		if err != nil {
-			return fmt.Errorf("could not detect workspaces: %w", err)
-		}
-
 		globalHash, err := calculateGlobalHash(rootpath, config.RootPackageJSON, config.TurboJSON.Pipeline, config.TurboJSON.GlobalDependencies, c.PackageManager, config.Logger, os.Environ())
 		if err != nil {
 			return fmt.Errorf("failed to calculate global hash: %v", err)
 		}
 
 		c.GlobalHash = globalHash
-		// We will parse all package.json's simultaneously. We use a
-		// wait group because we cannot fully populate the graph (the next step)
-		// until all parsing is complete
-		parseJSONWaitGroup := new(errgroup.Group)
-		justJsons := make([]string, 0, len(spaces))
-		for _, space := range spaces {
-			justJsons = append(justJsons, filepath.Join(space, "package.json"))
-		}
 
-		f, err := globby.GlobFiles(rootpath, justJsons, getWorkspaceIgnores())
+		// Get the workspaces from the package manager.
+		workspaces, err := c.PackageManager.GetWorkspaces(rootpath)
+
 		if err != nil {
-			return err
+			return fmt.Errorf("workspace configuration error: %w", err)
 		}
 
-		for _, val := range f {
-			relativePkgPath, err := filepath.Rel(rootpath, val)
+		// We will parse all package.json's simultaneously. We use a
+		// wait group because we cannot fully populate the graph (the next step)
+		// until all parsing is complete
+		parseJSONWaitGroup := new(errgroup.Group)
```

Comment:
Minor, but since you touched the code, prefer `&myStruct{}` over `new(myStruct)`. https://github.com/uber-go/guide/blob/master/style.md#initializing-struct-references

Reply:
Copy/paste! I just moved the line! /me wanders off to make the change.